File: Algo_Method/multiple_algo/Q2-1/Q2-1.py
def f(x): return x * (x * (x + 1) + 2) + 3

n = int(input())
l = -1
r = n + 1
while r - l > 0.001:
    mid = (r + l) / 2.0
    if f(mid) > n: r = mid
    else: l = mid
print(l)

File: Algo_Method/multiple_algo/Q2-1/Q2-1.rb
def f(x) return x * (x * (x + 1) + 2) + 3
end 

# INPUT
n = gets.chomp.to_i

l = -1
r = 101
while r - l > 0.001 do
    mid = (l + r) / 2.0
    if f(mid) > n
        r = mid
    else
        l = mid
    end
end
p l